packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js

```diff
@@ -100,15 +100,6 @@ function preloadHeadersByPage({ pages, manifest, pathPrefix, publicFolder }) {
   const appDataPath = publicFolder(PAGE_DATA_DIR, `app-data.json`)
   const hasAppData = existsSync(appDataPath)
 
-  let hasPageData = false
-  if (pages.size) {
-    // test if 1 page-data file exists, if it does we know we're on a gatsby version that supports page-data
-    const pageDataPath = publicFolder(
-      getPageDataPath(pages.get(pages.keys().next().value).path)
-    )
-    hasPageData = existsSync(pageDataPath)
-  }
-
```

Comment:
This heuristic is no longer working as intended with non-ssg pages and it was there to support really old gatsby version (early v2?)

Reply:
Indeed